Tests/SonarScanner.MSBuild.Tasks.UnitTests/GetAnalyzerSettingsTests.cs

```diff
@@ -219,15 +222,27 @@ public void GetAnalyzerSettings_ConfigExists_NewBehaviour_SettingsMerged()
 
             // Arrange
             // SONARMSBRU-216: non-assembly files should be filtered out
-            var filesInConfig = new List<string>
+            var filesInConfig = new List<AnalyzerPlugin>
             {
-                "c:\\config\\analyzer1.DLL",
-                "c:\\config\\analyzer2.dll",
-                "c:\\not_an_assembly.exe",
-                "c:\\not_an_assembly.zip",
-                "c:\\not_an_assembly.txt",
-                "c:\\not_an_assembly.dll.foo",
-                "c:\\not_an_assembly.winmd"
+                new AnalyzerPlugin
+                {
+                    AssemblyPaths = new List<string>
+                    {
+                        "c:\\config\\analyzer1.DLL",
+                        "c:\\not_an_assembly.exe",
+                        "c:\\not_an_assembly.zip",
+                    }
+                },
+                new AnalyzerPlugin
+                {
+                    AssemblyPaths = new List<string>
+                    {
+                        "c:\\config\\analyzer2.dll",
+                        "c:\\not_an_assembly.txt",
+                        "c:\\not_an_assembly.dll.foo",
+                        "c:\\not_an_assembly.winmd"
+                    }
+                }
```

Comment:
I deliberately split the previous file list across multiple plugins to make sure the filtering is applied across the whole set of plugins.